File: frontend/src/Components/Requests/LogoutRequest.js
import server from "../Static/Constants";

const LogoutRequest = async () => {
  try {
    const response = await fetch(`${server}/logout/`, {
      method: "PATCH",
      credentials: "include",
    });

    if (response.ok) {
      console.log("Logged out successfully");
      localStorage.removeItem("token");
      document.cookie =
        "token=; expires=Thu, 01 Jan 1970 00:00:00 UTC; path=/;";
      window.location.href = "/login/";
    } else {
      console.error("Failed to logout");
    }
  } catch (error) {
    console.error("Error during logout:", error);
  }
};

export default LogoutRequest;

File: flutter_frontend/lib/assets/auth/requests/logoutRequest.dart
import 'package:http/http.dart' as http;
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:flutter_frontend/static/constants.dart';

final FlutterSecureStorage secureStorage = const FlutterSecureStorage();

Future<void> logoutRequest() async {
  final response = await http.patch(Uri.parse('$server/logout/'));

  if (response.statusCode >= 200 && response.statusCode < 300) {
    await secureStorage.delete(key: 'token');
  }
}
